### ISSUE TITLE
Possibility of a library API?

### ISSUE THREAD

**Issue by Plecra** [chmln/handlr#27]
It's be really great to have a library that could be used for any application that wants to generically "open" files, or register itself as a default file handler. Since this would include a lot of the logic in `handlr`, would you have any interest in creating this API?

**Comment by Plecra** [comment]
Rust libraries are generally expected to be cross platform where the API isn't tied to a specific OS. It looks like most of the APIs would be implemented similarly using the Windows registry

**Comment by chmln** [comment]
@Plecra Yeah that's a good idea and I'm definitely open to it, let me know if you have any specific ideas :)
